### ISSUE TITLE
Phoenicis crashes when running from a Wayland session

### ISSUE THREAD

**Issue by ThanosApostolou** [PhoenicisOrg/phoenicis#1698]
### Expected behavior
Program should run normally on a Gnome Wayland session (probably using xwayland).

### Actual behavior
Running phoenicis crashes (core dumped) with output:
```
(PhoenicisPlayOnLinux:13598): Gdk-CRITICAL **: 01:33:24.348: gdk_x11_display_set_window_scale: assertion 'GDK_IS_X11_DISPLAY (display)' failed
#
# A fatal error has been detected by the Java Runtime Environment:
#
#  SIGSEGV (0xb) at pc=0x0000000000000002, pid=13598, tid=13636
#
# JRE version: OpenJDK Runtime Environment (11.0.1+13) (build 11.0.1+13)
# Java VM: OpenJDK 64-Bit Server VM (11.0.1+13, mixed mode, tiered, compressed oops, g1 gc, linux-amd64)
# Problematic frame:
# C  0x0000000000000002
#
# Core dump will be written. Default location: Core dumps may be processed with "/usr/lib/systemd/systemd-coredump %P %u %g %s %t %c %h %e" (or dumping to /usr/share/phoenicis/app/core.13598)
#
# An error report file with more information is saved as:
# /tmp/hs_err_pid13598.log
#
# If you would like to submit a bug report, please visit:
#   http://bugreport.java.com/bugreport/crash.jsp
# The crash happened outside the Java Virtual Machine in native code.
# See problematic frame for where to report the bug.
#
/usr/bin/phoenicis: line 3: 13598 Aborted                 (core dumped) /usr/share/phoenicis/PhoenicisPlayOnLinux "$@"

```

The **hs_err_pid13598.log** is here: https://pastebin.com/2y1TXwei

### Steps to reproduce
1. Use Gnome Wayland session in some linux distribution
2. Run phoenicis

### System information
* software version (git revision) of phoenicis and scripts repositories:
I tried version **alpha2** both with the provided linux tar.gz and with building the package myself
* used operating system:
Arch linux
* used Java version:
Openjdk 11.0.1

**Comment by qparis** [comment]
Hum interesting. Thank you.

Can you try to export this environnement variable before running Phoenicis?

export GDK_BACKEND=x11

**Comment by ImperatorS79** [comment]
Javafx does not work on Xwayland, unless you use the workaround here [https://github.com/javafxports/openjdk-jfx/issues/175](https://github.com/javafxports/openjdk-jfx/issues/175). The bug is solved in openjfx12.

**Comment by ThanosApostolou** [comment]
You are right, it works fine with `GDK_BACKEND=x11`.

I patched the launcher (https://aur.archlinux.org/cgit/aur.git/plain/fix-launcher-on-wayland.patch?h=phoenicis-playonlinux) at my AUR package (https://aur.archlinux.org/packages/phoenicis-playonlinux/) in order to make it work. I don't know if you want to add it at official builds too.

**Comment by plata** [comment]
This variable does not have any effect if you're on X11 anyways, right?

**Comment by ThanosApostolou** [comment]
From what I know, it doesn't have any effect on x11. I tested it and it works fine on both Wayland and Xorg Gnome sessions.

**Comment by plata** [comment]
Ok, then we should set it. The question is if we can find a smart solution such that it is used everywhere (I don't think it's possible to do this in Java).

**Comment by qparis** [comment]
In fact on OSX, we need to override some environment variables to extend the PATH. Other cases may happen. 
I suggest we make phoenicis_build_package create a launcher script that setup-up required environement variables.
I think we need to include a launcher script inside the officials.

**Comment by plata** [comment]
Ok.

**Comment by madoar** [comment]
I think this has been fixed in the latest JavaFX release `11.0.2`

**Comment by plata** [comment]
@ThanosApostolou if you still have the issue, feel free to re-open.
